src/ape_accounts/accounts.py

```diff
@@ -162,15 +162,31 @@ def delete(self):
         self.keyfile_path.unlink()
 
     def sign_message(self, msg: Any, **signer_options) -> Optional[MessageSignature]:
+        sign_raw_hash = False
+
         if isinstance(msg, str):
-            user_approves = self.__autosign or click.confirm(f"Message: {msg}\n\nSign: ")
+            display_msg = f"Signing raw string: '{msg}'"
             msg = encode_defunct(text=msg)
+
         elif isinstance(msg, int):
-            user_approves = self.__autosign or click.confirm(f"Message: {msg}\n\nSign: ")
+            display_msg = f"Signing raw integer: {msg}"
             msg = encode_defunct(hexstr=HexBytes(msg).hex())
+
         elif isinstance(msg, bytes):
-            user_approves = self.__autosign or click.confirm(f"Message: {msg.hex()}\n\nSign: ")
-            msg = encode_defunct(primitive=msg)
+            display_msg = f"Signing raw bytes: '{msg.hex()}'"
+
+            if sign_raw_hash := (
+                len(msg) == 32 and not click.confirm("Sign using EIP-191? (Select 'y' if unsure)")
```

Comment:
Using `click`-isms outside of a CLI is a bit strange, I know we are already doing it elsewhere in this calss, but it'd be nice to keep prompting in the CLI components.

Reply:
This class needs to prompt the user directly to unlock or approve actions, so seems okay to me